lib/qmp.ml

```diff
@@ -215,10 +256,18 @@ let message_of_string str =
       let fdset_id = json |> U.member "fdset-id" |> U.to_int in
       {fd; fdset_id}
     in
+    let qom json =
+      let mem_of k x = x |> U.member k |> U.to_string in
+      json |> U.convert_each (fun x -> {name = (mem_of "name" x);ty = (mem_of "type" x)})
+    in
     let result =
       let return = json |> U.member "return" in
       return |> function
-      | `List  _ -> return |> name_list |> fun x -> Name_list x
+      | `List  _ -> (return |> U.convert_each U.keys |> List.flatten |> function
+        | x when (subset_of ["name"; "type"] x) -> return |> qom |> fun x -> Qom x
```

Comment:
Sorry, I just realised that using `x |> subset_of` you were being consistent with the style of the rest of the file.
Please use `x |> subset_of ["name"; "type"]` and `x |> subset_of ["name"]` as you did in the initial pull request

Reply:
Ok，np